modules/scripts/startup-script/files/setup-raid.yml

```diff
@@ -0,0 +1,96 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+
+- name: Configure local SSDs
+  become: true
+  hosts: localhost
+  vars:
+    DST_MNT: "/mnt/localssd"
+    bashrc: "{{ '/etc/bashrc' if ansible_facts['os_family'] == 'RedHat' else '/etc/bash.bashrc' }}"
+    array_dev: "/dev/md0"
+    mode: 755
+  tasks:
+  - name: Check if DST_MNT exists
+    ansible.builtin.stat:
+      path: "{{ DST_MNT }}"
+    register: dst_mnt_exists
+
+  - name: Check if {{ array_dev }} exists
+    ansible.builtin.stat:
+      path: "{{ array_dev }}"
+    register: array_dev_exists
+
+  - name: Install mdadm
+    ansible.builtin.package:
+      name: mdadm
+      state: present
+
+  - name: Get number of devices
+    ansible.builtin.shell: nvme list | { grep nvme_ || test $? = 1; }  | { grep -v nvme_card-pd || test $? = 1; } | awk '{print $1}' | wc -l
+    register: nvme_device_count
+
+  - name: Get list of devices
+    ansible.builtin.shell: nvme list | { grep nvme_ || test $? = 1; }  | { grep -v nvme_card-pd || test $? = 1; } | awk '{print $1}' | paste -sd ' '
+    register: nvme_device_list
+
+  - name: Exit if less than 2 local ssd
+    ansible.builtin.meta: end_play
+    when: nvme_device_count.stdout < "2"
+
+  - name: Check if there is an existing foreign RAID device
+    ansible.builtin.shell: lsblk | grep -q 'md127'
+    register: raid_check
+    failed_when: false
+    changed_when: false
+
+  - name: Stop and remove foreign RAID device
+    ansible.builtin.shell: mdadm --manage /dev/md127 --stop --remove
+    when: raid_check.rc == 0
+
+  - name: Zero superblock on foreign RAID device
+    ansible.builtin.shell:
+      cmd: sudo mdadm --zero-superblock {{ nvme_device_list.stdout }}
+    when: raid_check.rc == 0
+
+  - name: Create RAID-0 array if none were found
+    ansible.builtin.command: mdadm --create /dev/md0 --level=0 --raid-devices={{ nvme_device_count.stdout }} {{ nvme_device_list.stdout }}
+    when: not array_dev_exists.stat.exists
+
+  - name: Create ext4 filesystem
+    filesystem:
+      fstype: ext4
+      device: /dev/md0
+
+  - name: Tune reserved blocks
+    ansible.builtin.shell:
+      cmd: tune2fs /dev/md0 -r 131072
+
+  - name: Add a label
+    ansible.builtin.shell: e2label /dev/md0 LOCALSSD
+
+  - name: Create mount point
```

Comment:
You actually want to order "Create mount point" _after_ `ansible.builtin.mount` (and rename it to "set mount permissions" or similar). The defined behavior of `mounted` will create the mount point but you are causing the permissions to disappear by mounting over them.

https://docs.ansible.com/ansible/2.9/modules/mount_module.html

Reply:
Good catch. Changed the order